security/pkg/k8s/configutil.go

```diff
@@ -49,14 +49,14 @@ func InsertDataToConfigMap(client corev1.ConfigMapsGetter, lister listerv1.Confi
 		if _, err = client.ConfigMaps(meta.Namespace).Create(context.TODO(), configmap, metav1.CreateOptions{}); err != nil {
 			// Namespace may be deleted between now... and our previous check. Just skip this, we cannot create into deleted ns
 			// And don't retry a create if the namespace is terminating
-			if errors.IsNotFound(err) || errors.HasStatusCause(err, v1.NamespaceTerminatingCause) {
+			if errors.IsAlreadyExists(err) || errors.HasStatusCause(err, v1.NamespaceTerminatingCause) {
```

Comment:
This is the fix.

Reply:
Do we have a test for it?

Reply:
This is a common usage of client-go. I am not sure if there is a test for all these kinds of code.

Reply:
I understand but how do we ensure we captured the right error -  Can this test TestInsertDataToConfigMap be extended to have this case to ensure we are handling the right error

Reply:
TBH, i donot know how to test this. Even in k8s `Create` error values are not tested thoroughly.

Reply:
:-)

Reply:
Does the fake client not return proper errors? If it doesn't we should file an upstream request. If it does we should add a test

Reply:
WIll investigate on this